docs/06-contributors/999-rfcs/000-stories/story-15.md

```diff
@@ -0,0 +1,173 @@
+# User Story 15 - Task List with Api Gateway and Redis
+
+> **Status**: Draft, Expected released on 2023/03/16
+
+
+The following code is an inital implementation of TaskList with api gateway and a redis db 
+
+## New landuage and SDK features it introduces
+
+- [x] bring untyped
+  - [x] bring external npm package (axios)
+  - [x] bring an internal nodejs stdlib (RegEx)
+  - [x] How does untyped works with numeric operations?
+  - [x] Do we cast untyped? 
+- [x] enum & duration that can be included inside json
+- [x] It leverages setting explicit permissions (using the `this.inflight` API, described [here](https://github.com/winglang/wing/pull/1610))
+- [x] Using interface 
+- [x] Using await when calling a untyped function that returns a promise
+- [ ] bring cdktf
+- [ ] use redis instead of bucket
+- [ ] code that updates estimation and duration from REST post command
+
+## Discussion topics
+- `get` vs `on_get` - Should the `cloud.api` API have the `on_` prefix to match `cloud.bucket` API and also to allow calling
+the api get/post/delete/put commands (`api.get(url)` vs `api.on_get(path, inflight ())`
+- Routing params - I have used express's synatx for dynamic parts of the path for the api gateway `get("/task/:id"`, 
+but then I noticed that [aws](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-step-by-step.html) 
+uses the `{id}` syntax. What should be the the right syntax?
+- Bring internal nodejs types - how do we bring somethink that we can't require, like `RegEx`
+- From anything to Json - can I cast something from bring untyped to Json? 
+- Calling await 
+
+## Code 
+```ts (wing)
+
+bring cloud;
+
+// TODO discuss how we bring untyped something like RegEx from JavaScript 
+// PLACEHOLER for bringing something from Javascript stdlib
+bring untyped RegExp from_js("RegExp"); 
+bring untyped Math from_js("Math"); 
+
+
+// prerequisite: npm install axios
+// PALCEHOLDER for bringing some external module
+bring untyped axios require_from_js("axios"); 
+
+enum Status {
+  Uncompleted,
+  Completed
+}
+
+interface ITaskListModel {
+  inflight get(id: str): Json;
+  inflight add(title: str): str;
+  inflight remove(id: str): void; 
+  inflight find(r: RegExp): Array<str>;
+  inflight set_status(id: str, status: Status): str;
+  inflight set_estimation(id: str, estimation: duration): str;
+}
+
+resource TaskListModel implementes ITaskListModel {
+  _bucket: cloud.Bucket;
+  init() {
+    this._bucket = new cloud.Bucket();
+    this.inflights.add("get", ref: "this._bucket", op: "get");
```

Comment:
All of these should be implicit by the end of the sprint I believe. @yoav-steinberg ?

Reply:
If that is the case, I would work to add a non trivial example that can't be calculated by the compiler and describe how we want the compiler to behave. (error/warning/ default behavior)

Waiting for @yoav-steinberg